frontend/src/features/recommendation/SearchFilter.spec.tsx

```diff
@@ -20,16 +31,20 @@ describe('Filters feature', () => {
     // Some context is needed by the component SearchFilter
     render(
       <Router history={history}>
-        <ThemeProvider theme={theme}>
-          <SearchFilter />
-        </ThemeProvider>
+        <StyledEngineProvider injectFirst>
+          <ThemeProvider theme={theme}>
+            <SearchFilter />
+          </ThemeProvider>
+        </StyledEngineProvider>
       </Router>
     );
   });
 
   function clickOnShowMore(wasClosed = true) {
     const checkboxDisplayFilters = screen.queryByLabelText('show more');
     expect(checkboxDisplayFilters).not.toBeNull();
+    if (checkboxDisplayFilters == null)
```

Comment:
Si checkboxDisplayFilters == null, le test devrait s'arrêter à la ligne du dessus : "expect(checkboxDisplayFilters).not.toBeNull();", donc cette vérification n'est pas nécessaire

Reply:
lignes 91 et 114 non plus ça ne me paraît pas nécessaire

Reply:
My mistake, I did not realize that typescript was doing static type checks on test files. And that it doesn't realize that a failed expect would have immediately stopped the test.

Reply:
These typescript errors are supposed to be ignored in test files.  
It seems that a regression has been introduced in react-scripts v5: https://github.com/facebook/create-react-app/pull/11875